lib/common/ipc_attrd.c

```diff
@@ -497,3 +497,96 @@ pcmk__attrd_api_update(pcmk_ipc_api_t *api, const char *node, const char *name,
 
     return rc;
 }
+
+int
+pcmk__attrd_api_update_list(pcmk_ipc_api_t *api, GList *attrs, const char *dampen,
+                            const char *set, const char *user_name,
+                            uint32_t options)
+{
+    int rc = pcmk_rc_ok;
+    xmlNode *request = NULL;
+
+    if (attrs == NULL) {
+        return EINVAL;
+    }
+
+    /* There are two different ways of handling a list of attributes:
+     *
+     * (1) For messages originating from some command line tool, we have to send
+     *     them one at a time.  In this loop, we just call pcmk__attrd_api_update
+     *     for each, letting it deal with creating the API object if it doesn't
+     *     already exist.
+     *
+     *     The reason we can't use a single message in this case is that we can't
+     *     trust that the server supports it.  Remote nodes could be involved
+     *     here, and there's no guarantee that the version of attrd running on a
+     *     remote node will be recent enough.  We also can't just try sending a
```

Comment:
attrd doesn't run on remote nodes ... there's no guarantee that a newer client running on a remote is talking to (proxied through) a cluster node with a newer attrd

Reply:
Fixed.